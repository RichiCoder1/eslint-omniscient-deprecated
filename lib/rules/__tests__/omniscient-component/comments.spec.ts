import { fix } from "./__setup__/setup";

describe("comments", () => {
    it("single line comments should be included", () => {
        const input = `
import component from 'omniscient';

const Test = component({
    // This is a
    a: 1
}, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("block comments should be included", () => {
        const input = `
import component from 'omniscient';

const Test = component({
    /*
     * Block!
     */
    a: 1
}, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });

    it("mix and match comments should be included", () => {
        const input = `
import component from 'omniscient';

const Test = component({
    // This is a
    /*
     * Block!
     */
    a: 1
}, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
    it("trailing single line comments should be included", () => {
        const input = `
import component from 'omniscient';

const Test = component({

    a: 1 // This is a
}, () => {})`;
        const report = fix(input);
        expect(report).toBeFixed();
        expect(report).toMatchSnapshot();
    });
});
