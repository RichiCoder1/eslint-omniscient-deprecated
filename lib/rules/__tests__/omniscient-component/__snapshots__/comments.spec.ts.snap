// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comments block comments should be included 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class Test extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    /*
     * Block!
     */
    a = 1;

    render() {}
}",
}
`;

exports[`comments mix and match comments should be included 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class Test extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    // This is a
    /*
     * Block!
     */
    a = 1;

    render() {}
}",
}
`;

exports[`comments single line comments should be included 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class Test extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    // This is a
    a = 1;

    render() {}
}",
}
`;

exports[`comments trailing single line comments should be included 1`] = `
Object {
  "fixed": true,
  "messages": Array [],
  "output": "import { Component } from 'react';
import { isEqualImmutable } from 'ImmutableComponent';

import component from 'omniscient';

class Test extends Component {
    shouldComponentUpdate(nextProps, nextState) {
        return !(
            isEqualImmutable(this.props, nextProps) &&
            ((!oldState && !newState) ||
                isEqualImmutable(this.state, nextState))
        );
    }

    a = 1; // This is a

    render() {}
}",
}
`;
